### DISCUSSION TITLE
How is the OAUTH token secured?

### DISCUSSION THREAD

**Discussion by cmsolomon** [abraunegg/onedrive#2460] [Q&A] [answered]
Hi,
I am trying to get the security team at my company to authorise this client for use with our company accounts.
The one query they have come back with is how the OAUTH tokens are secured - their concern was that if someone compromised a users system an stole the token they would be able to gain ongoing access to that user's onedrive.

I checked the security documentation, but can see anything about that. Is the token just stored in plain text in the refresh_token file?

**Comment by abraunegg** [comment, marked as answer]
@cmsolomon 
> I checked the security documentation, but can see anything about that. Is the token just stored in plain text in the refresh_token file?

Correct - the token is stored in the users home directory, in the `.config` folder, with `0600` permissions - which `rw` for that user only, and needs to be `rw` as the token gets refreshed during application use.

> their concern was that if someone compromised a users system an stole the token they would be able to gain ongoing access to that user's onedrive.

Only if that token was still viable based on its TTL & refresh status state (known by OneDrive).

The only security mitigation here is user home directory encryption - this then prevents access of the users home directory if the decryption key is not known, thus that token cannot be accessed - but if they do - then the TTL of the token comes into play.

**Reply by cmsolomon** [reply]
Thanks for the quick response!

**Reply by spalfy** [reply]
Hi @abraunegg, do you plan to implement encryption for the refresh_token file, similar to what rclone does?

**Reply by abraunegg** [reply]
@spalfy 
By default 'rclone' does not *encrypt* any refresh token ... and actually stores more data in an un-ecrypted manner ......... so by this client only storing the 'refresh_token' with safe user permissions, this is already more secure than 'rclone' ..............

Now .. 'rclone' can go and secure the *entire* config file via the optional 'rclone config' command needing a password to decrypt the config file.

At this point in time, there is zero plan to do this sort of configuration change and/or change how this client operates.
